### PR TITLE
Add EditTaskCommand class and Update EditModuleCommand class

### PR DESCRIPTION
**Pull request by wangqinNick**

### CODE REVIEW

**Review by Aseanseen — APPROVED** [review]
Looks good to merge!

File: src/main/java/seedu/duke/data/TaskManager.java [hunk 1]
```diff
@@ -28,18 +28,18 @@ public static Task getTask(int taskId) throws TaskNotFoundException {
...
-     *  The edited task
      * @param taskId
      *  The index of the task in the Task List.
+     * @param newTaskDescription
+     *  The new description of the task
      * @throws TaskNotFoundException
      *  If
      */
-    public static void edit(Task editedTask, int taskId) throws TaskNotFoundException {
+    public static void edit(int taskId, String newTaskDescription) throws TaskNotFoundException {
```
Comment (SimJunYou) [inline src/main/java/seedu/duke/data/TaskManager.java]: TaskManager's edit function takes a new Task object to get around needing an edit function for each property of task, especially when we will be adding new task properties in the future. Is it possible to reverse this change here?

**Review by SimJunYou — CHANGES_REQUESTED** [review]
One conflicting issue noted below

File: src/main/java/seedu/duke/command/edit/EditTaskCommand.java [hunk 2]
```diff
@@ -1,23 +1,53 @@
...
+import static seedu.duke.util.ExceptionMessage.MESSAGE_TASK_NOT_FOUND;
+import static seedu.duke.util.Message.MESSAGE_EDIT_TASK_SUCCESS;
+
 
 public class EditTaskCommand extends EditCommand {
 
-    public EditTaskCommand(int taskIndex,String newTaskDescription){
+    private int taskID;
+    private String newTaskDescription;
+    public static final String COMMAND_WORD = "edt";
```
Comment (Aseanseen) [inline src/main/java/seedu/duke/command/edit/EditTaskCommand.java]: I believe the command should be changed to "edit -t"

**Review by Aseanseen — CHANGES_REQUESTED** [review]
Need amends to the command, e.g. "edit -t", "edit -m"

File: src/main/java/seedu/duke/command/edit/EditTaskCommand.java (hunk 2 above)
Comment (tobiasceg) [inline src/main/java/seedu/duke/command/edit/EditTaskCommand.java]: Is it possible to also add a EditTaskCommand format like ur EditModuleCommand

File: src/main/java/seedu/duke/data/TaskManager.java (hunk 1 above)
Comment (wangqinNick) [reply src/main/java/seedu/duke/data/TaskManager.java]: ok

File: src/main/java/seedu/duke/command/edit/EditTaskCommand.java (hunk 2 above)
Comment (wangqinNick) [reply src/main/java/seedu/duke/command/edit/EditTaskCommand.java]: done

Comment (wangqinNick) [reply src/main/java/seedu/duke/command/edit/EditTaskCommand.java]: done

File: src/main/java/seedu/duke/command/edit/EditTaskCommand.java [hunk 3]
```diff
@@ -1,23 +1,46 @@
...
+import static seedu.duke.util.Message.MESSAGE_EDIT_TASK_SUCCESS;
+
 
 public class EditTaskCommand extends EditCommand {
 
-    public EditTaskCommand(int taskIndex,String newTaskDescription){
+    private int taskID;
+    private String newTaskDescription;
+    public static final String COMMAND_WORD = Parser.COMMAND_WORD_EDIT + "-t";
+    public static final String FORMAT = Parser.COMMAND_WORD_EDIT + "-t" + " <task ID> <new task description>";
```
Comment (Aseanseen) [inline src/main/java/seedu/duke/command/edit/EditTaskCommand.java:17]: Maybe append with COMMAND_WORD here instead

File: src/main/java/seedu/duke/data/storage/Decoder.java [hunk 4]
```diff
@@ -30,17 +30,8 @@
...
-     * @param dataFileName
-     *  The name of the file to read from
-     * @return
-     *  The HashMap modulesMap
-     * @throws FileNotFoundException
-     *  When the file does not exist
-     */
-    public static HashMap<String, Module> loadModules(String dataFileName) throws FileNotFoundException {
+
+    public static HashMap<String, Module> loadModules(String dataFileName) {
```
Comment (Aseanseen) [inline src/main/java/seedu/duke/data/storage/Decoder.java:34]: Curious why javadoc deleted? 1 more occurence below

**Review by Aseanseen — APPROVED** [review]
Some suggestions but otherwise ok to merge

File: src/main/java/seedu/duke/data/storage/Decoder.java (hunk 4 above)
Comment (wangqinNick) [reply src/main/java/seedu/duke/data/storage/Decoder.java:34]: I deleted the exception throw. it is handled inside the func already

**Pull request merged**